### PR TITLE
Remove FileSystemAbstraction fom CommonAbstractStore

### PR DESCRIPTION
**Pull request by ragadeeshu**
Along with StoreChannel/FileChannel

### CODE REVIEW

File: community/kernel/src/main/java/org/neo4j/kernel/impl/store/CommonAbstractStore.java [hunk 1]
```diff
@@ -794,9 +723,15 @@ public void setHighestPossibleIdInUse( long highId )
 
     private long calculateHighestIdInUseByLookingAtFileSize()
     {
-        try
+        try ( PagedFile pagedFile = pageCache
+                .map( getStorageFileName(), pageCache.pageSize() - pageCache.pageSize() % getRecordSize() ) )
```
Comment (chrisvest) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/store/CommonAbstractStore.java]: Extract some variables so you don't need to break expressions across lines.

File: community/kernel/src/main/java/org/neo4j/kernel/impl/store/AbstractDynamicStore.java [hunk 2]
```diff
@@ -254,11 +255,21 @@ public int getNumberOfReservedLowIds()
     @Override
     protected void readAndVerifyBlockSize() throws IOException
```
Comment (davidegrohmann) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/store/AbstractDynamicStore.java:255]: This is not important for this PR but it might be in the long plan of cleaning up the store hierarchy: this method is only used in `https://github.com/neo4j/neo4j/blob/2.3/community/kernel/src/main/java/org/neo4j/kernel/impl/store/CommonAbstractStore.java#L210` so it should be made private and moved there. This can be of course be part of another PR.

File: community/kernel/src/main/java/org/neo4j/kernel/impl/store/CommonAbstractStore.java [hunk 3]
```diff
@@ -131,7 +121,7 @@ public CommonAbstractStore(
                     e.addSuppressed( failureToClose );
                 }
             }
-            closeFileChannel();
+            close();
```
Comment (davidegrohmann) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/store/CommonAbstractStore.java]: Here you are closing `storeFile` twice one time before calling `close` and the second time in `close()`. It smells a little bit I think it should be cleaned up.

Moreover this looks like a change of the current behavior in case of failure when loading the store. Do we have enough test coverage to ensure that we are not messing up?

File: community/kernel/src/main/java/org/neo4j/kernel/impl/store/RelationshipGroupStore.java [hunk 4]
```diff
@@ -124,12 +124,21 @@ public int getNumberOfReservedLowIds()
...
+                if ( pageCursor.next() )
+                {
+                    do
+                    {
+                        denseNodeThreshold = pageCursor.getInt();
+                    }
+                    while ( pageCursor.shouldRetry() );
+                }
+            }
+        }
```
Comment (davidegrohmann) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/store/RelationshipGroupStore.java]: Isn't this the same code as in `AbstractDynamicStore.readAndVerifyBlockSize()`?

File: community/kernel/src/main/java/org/neo4j/kernel/impl/store/AbstractDynamicStore.java (hunk 2 above)
Comment (ragadeeshu) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/store/AbstractDynamicStore.java:255]: The point of this method is that it is implemented differently in `AbstractStore`, `AbstractDynamicStore` and `RelationshipGroupStore`.
